concent_api/core/payments/sci_backend.py

```diff
@@ -0,0 +1,140 @@
+    amount_pending  = subtasks_price - amount_paid
+
+    return (amount_paid, amount_pending)
+
+
+def get_clients_eth_accounts(task_to_compute: message.tasks.TaskToCompute):
+    assert isinstance(task_to_compute, message.tasks.TaskToCompute) and not None
+
+    requestor_eth_address   = task_to_compute.requestor_ethereum_public_key
+    provider_eth_address    = task_to_compute.provider_ethereum_public_key
+
+    return (requestor_eth_address, provider_eth_address)
+
+
+def get_number_of_eth_block(concent_rpc: implementation.SCIImplementation):
+    assert isinstance(concent_rpc, implementation.SCIImplementation) and not None
+
+    return concent_rpc.get_block_number()
+
+
+def get_list_of_payments(
+    concent_rpc:            implementation.SCIImplementation    = None,
+    requestor_eth_address:  str                                 = None,
+    provider_eth_address:   str                                 = None,
+    payment_ts:             int                                 = None,
+    current_time:           int                                 = None,
+    to_block:               int                                 = None,
+    transaction_type:       str                                 = None,
+):
+    """
+    Function which return list of transactions from payment API
+    where timestamp >= T0
+    """
+    assert isinstance(concent_rpc,              implementation.SCIImplementation)   and not None
+    assert isinstance(requestor_eth_address,    str)                                and len(requestor_eth_address) == 42
+    assert isinstance(provider_eth_address,     str)                                and len(provider_eth_address)  == 42
+    assert isinstance(payment_ts,               int)                                and payment_ts     > 0
+    assert isinstance(current_time,             int)                                and current_time   > 0
+    assert isinstance(to_block,                 int)                                and to_block       > 0
+    assert isinstance(transaction_type,         str)                                and transaction_type in ['Force', 'Batch']
+
+    chain_segment_amount    = math.ceil((current_time - payment_ts) / settings.CREATION_NEW_CHAIN_SEGMENT_TIME)
+    from_block              = to_block - chain_segment_amount
+
+    if transaction_type == 'Force':
+        payments_list = concent_rpc.get_forced_payments(
+            requestor_address   = requestor_eth_address,
+            provider_address    = provider_eth_address,
+            from_block          = from_block,
+            to_block            = to_block,
+        )
+    elif transaction_type == 'Batch':
+        payments_list = concent_rpc.get_batch_transfers(
+            payer_address   = requestor_eth_address,
+            payee_address   = provider_eth_address,
+            from_block      = from_block,
+            to_block        = to_block,
+        )
+    else:
+        return None
```

Comment:
You already asserted that transaction_type is either Force or Batch, so this code will not be ever reached.

Reply:
Done